python/tvm/octo/compile.py

```diff
@@ -184,7 +192,20 @@ def compile(
     # Perform legalization to lower Relax operators.
     relax_mod = relax.transform.LegalizeOps()(relax_mod)
 
-    # Schedule all remaining functions to be compatible with gpu if needed.
+    # If specified, perform tuning to optimize remaining workloads.
+    if tuning_steps:
+        with tempfile.TemporaryDirectory() as work_dir:
+            with target, tvm.transform.PassContext(trace=Trace(relax_mod), opt_level=0):
+                tuning_pass = relax.transform.MetaScheduleTuneIRMod(
+                    params={},
+                    work_dir=work_dir,
+                    max_trials_global=tuning_steps,
+                )
+                relax_mod = tuning_pass(relax_mod)
+                application_pass = relax.transform.MetaScheduleApplyDatabase(work_dir)
+                relax_mod = application_pass(relax_mod)
+
+    # Finally, add thread binding to remaining kernels to allow them to run on gpu.
```

Comment:
I think we only skipped those kernels where it's thread-binded already, which is the same case for DefaultGPUSchedule pass where it also skips these binded kernels. Therefore, I think it's safe to apply either MS tuning or default schedule, no need to apply both here.

Reply:
I was thinking that if someone picked very very few trials, then not all kernels would be tuned. We'd need to bind the remaining kernels with DefaultScheduleGPU or itd error out.

Reply:
I see, that make sense to me.